api/src/indigo_structure_checker.h

```diff
@@ -0,0 +1,86 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#ifndef __indigo_structure_checker__
+#define __indigo_structure_checker__
+
+#include "base_c/defs.h"
+#include "base_cpp/exception.h"
+#include "indigo_internal.h"
+#include "molecule/structure_checker2.h"
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+
+namespace indigo
+{
+
+
+    class DLLEXPORT IndigoStructureChecker : public StructureChecker2
+    {
+    public:
+        /// <summary>
+        /// Textual Check Type values for the corresponding CheckTypeCode
+        /// to be used in host language calls (Java, Python etc.) as a comma-separated list
+        /// See below check(char *, char *):
+        /// check("molecule text", "load, valence, radical, atoms 1 2 3, bonds 4 5 6, tgroup"
+        /// is equivalent to
+        /// check(the_molecule, check_flags = CHECK_LOAD | CHECK_VALECE | CHECK_RADICAL | CHECK_TGROUP, selected_atoms=[1,2,3], selected_bonds=[4,5,6]);
+        /// </summary>
+        static constexpr char* CHECK_NONE_TXT = "none";                 // Check none
```

Comment:
Code duplicate for the check types. It is better to keep all the check types in structure_checker2. Then remove all this strings

Reply:
And use types_map<CheckMessageCode , string>